test/jdk/java/security/cert/CertPathBuilder/selfIssued/KeyUsageMatters.java

```diff
@@ -179,56 +184,42 @@ public final class KeyUsageMatters {
 
     private static Set<TrustAnchor> generateTrustAnchors()
             throws CertificateException {
-        // generate certificate from cert string
-        CertificateFactory cf = CertificateFactory.getInstance("X.509");
 
-        ByteArrayInputStream is =
-                    new ByteArrayInputStream(selfSignedCertStr.getBytes());
-        Certificate selfSignedCert = cf.generateCertificate(is);
+        // generate certificate from cert string
+        X509Certificate selfSignedCert = PEM_DECODER.decode(selfSignedCertStr, X509Certificate.class);
 
         // generate a trust anchor
         TrustAnchor anchor =
-            new TrustAnchor((X509Certificate)selfSignedCert, null);
+            new TrustAnchor(selfSignedCert, null);
 
         return Collections.singleton(anchor);
     }
 
     private static CertStore generateCertificateStore() throws Exception {
-        Collection entries = new HashSet();
+        Collection<DEREncodable> entries = new HashSet<>();
 
         // generate certificate from certificate string
-        CertificateFactory cf = CertificateFactory.getInstance("X.509");
-
-        ByteArrayInputStream is;
-
-        is = new ByteArrayInputStream(targetCertStr.getBytes());
-        Certificate cert = cf.generateCertificate(is);
+        DEREncodable cert = PEM_DECODER.decode(targetCertStr, X509Certificate.class);
```

Comment:
nit:  there is little value storing the decode() in a variable that is then added to `entries()`. You could do `entries.add(PEM_DECODER...)`